hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCommitArchiveLog.java

```diff
@@ -173,6 +189,49 @@ private boolean deleteArchivedInstants(List<HoodieInstant> archivedInstants) {
         throw new HoodieIOException("Failed to delete archived instant " + archivedInstant, e);
       }
     }
+
+    // Remove older meta-data from auxiliary path too
+    Optional<HoodieInstant> youngestCommitted =
+        archivedInstants.stream()
+            .filter(i -> {
+              return i.isCompleted()
+                  && (i.getAction().equals(COMMIT_ACTION) || (i.getAction().equals(DELTA_COMMIT_ACTION)));
+            })
+            .sorted(Comparator.comparing(HoodieInstant::getTimestamp).reversed()).findFirst();
+    if (youngestCommitted.isPresent()) {
+      success &= deleteAllInstantsOlderorEqualsInAuxMetaFolder(youngestCommitted.get());
+    }
+    return success;
+  }
+
+  /**
+   * Remove older instants from auxiliary meta folder
+   *
+   * @param instant Hoodie Instant
+   * @return success if all eligible file deleted successfully
+   * @throws IOException in case of error
+   */
+  private boolean deleteAllInstantsOlderorEqualsInAuxMetaFolder(HoodieInstant instant)
+      throws IOException {
+    List<HoodieInstant> instants =
+        HoodieTableMetaClient.scanHoodieInstantsFromFileSystem(metaClient.getFs(),
+            new Path(metaClient.getMetaAuxiliaryPath()),
+            HoodieActiveTimeline.INCLUDED_EXTENSIONS_IN_ACTIVE_TIMELINE);
+
+    List<HoodieInstant> instantsToBeDeleted =
+        instants.stream().filter(instant1 -> {
+          return instant1.getTimestamp().compareTo(instant.getTimestamp()) <= 0;
```

Comment:
is nt there a method to compare instants to each other more nicely?

Reply:
Fixed.